### PR TITLE
Support nested vector in explorer

### PR DESCRIPTION
**Pull request by BriungRi**
Use `JSON.parse` to handle parsing `[]` arguments. 

Previously, `[[abc]]` would be parsed as `["[abc]"], but really, it should be `[["abc"]]`.

The noteworthy change here is that quotes must now be used for literals. E.g. `[0xabc]` used to be acceptable but now must be specified as `["0xabc"]`

### CODE REVIEW

File: src/pages/Account/Tabs/ModulesTab/Contract.tsx [hunk 1]
```diff
@@ -277,6 +277,7 @@ function RunContractForm({
         const type = fnParams[i];
         if (type.includes("vector")) {
           // when it's a vector<u8>, we support both hex and javascript array format
+          console.log(type, arg);
```
Comment (gregnazario) [inline src/pages/Account/Tabs/ModulesTab/Contract.tsx:280]: Remove console.log?

File: src/utils.ts [hunk 2]
```diff
@@ -180,16 +180,36 @@ export function encodeInputArgsForViewRequest(type: string, value: string) {
...
-  if (result[0] === "[" && result[result.length - 1] === "]") {
-    result = result.slice(1, -1);
+// Also works nested [[1, 2], [3, 4]] to [["1", "2"], ["3", "4"]]
+// TODO: Need to type this recursively
+export function deserializeVector(vectorString: string): any {
+  const trimmed = vectorString.trim();
+  let inside = "";
+  if (isArrayStr(trimmed)) {
+    inside = trimmed.slice(1, -1);
   }
```
Comment (gregnazario) [inline src/utils.ts]: What's this doing?

Can we put a comment above it?

File: src/utils.ts [hunk 3]
```diff
@@ -180,16 +180,36 @@ export function encodeInputArgsForViewRequest(type: string, value: string) {
 }
 
 // Deserialize "[1,2,3]" or "1,2,3" to ["1", "2", "3"]
-export function deserializeVector(vectorString: string): string[] {
-  let result = vectorString.trim();
-  if (result[0] === "[" && result[result.length - 1] === "]") {
-    result = result.slice(1, -1);
+// Also works nested [[1, 2], [3, 4]] to [["1", "2"], ["3", "4"]]
+// TODO: Need to type this recursively
```
Comment (gregnazario) [inline src/utils.ts]: Isn't this recursive now?

File: src/pages/Account/Tabs/ModulesTab/Contract.tsx [hunk 4]
```diff
@@ -287,6 +288,7 @@ function RunContractForm({
         } else return arg;
       }),
     };
+    console.log(payload);
```
Comment (kent-white) [inline src/pages/Account/Tabs/ModulesTab/Contract.tsx:291]: logss

**Review by kent-white — COMMENTED** [review]
logs, otherwise this looks good :)

File: src/utils.ts [hunk 5]
```diff
@@ -180,16 +180,36 @@ export function encodeInputArgsForViewRequest(type: string, value: string) {
...
+    let build = "";
+    for (let i = 0; i < inside.length; i++) {
+      if (inside[i] === "]") {
+        innerArrayStrs.push(build);
+        build = "";
+      } else {
+        build += inside[i];
+      }
+    }
+    return innerArrayStrs.map(deserializeVector);
```
Comment (gregnazario) [inline src/utils.ts:205]: This code will have the bracket at the beginning?

But, I guess you call it recursively?

I think this misses, can we make a unit test, I suspect that:

```
[[blah,blah],[blah,blah]]
```
Will get to:
```
[blah, blah], [blah, blah]
```
Then 
```
innerArrayStrs.push('[blah, blah')
innerArrayStrs.push(',[blah, blah')
```

Then do that recursively which will return two items

```
'[blah', 'blah'
'','[blah', 'blah'
```

**Review by kent-white — APPROVED** [review]

**Pull request merged**